android-app/src/main/java/org/example/app/di/UnitFactoryModule.kt

```diff
@@ -0,0 +1,16 @@
+package org.example.app.di
+
+import dagger.Binds
+import dagger.Module
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+import org.example.app.units.NewsListUnitsFactory
+import org.example.library.SharedFactory
+
+@InstallIn(SingletonComponent::class)
+@Module
+abstract class UnitFactoryModule {
+    @Binds
+    abstract fun bindNewsListUnitFactory(impl: NewsListUnitsFactory): SharedFactory.NewsUnitsFactory
```

Comment:
why here used `@Binds` instead of `@Provides`?

Reply:
Because we need to implement interface instance, and that can be done with @Binds annotation, there is an [guide](https://developer.android.com/training/dependency-injection/hilt-android#inject-interfaces) how to do this using Hilt.